### ISSUE TITLE
Nightcode fails to run on OSX

### ISSUE THREAD

**Issue by betabrain** [oakes/Nightcode#205]
I just installed Nightcode, but failed to get it to run. Here is the error message when calling Nightcode from the Terminal:

```
$ open -a Nightcode.app
LSOpenURLsWithRole() failed for the application /Applications/Nightcode.app with error -10810.
```

### Nightcode Version
2.1.6 (as linked on the website)

### OS Version
OS X 10.11.6 El Capitan

### Java Version
```
$ java -version
java version "1.8.0_101"

$ javac -version
javac 1.8.0_101
```

### Possibly Related
[Stack Overflow: Opening .app on Mac OS X 10.8.5 results in error -10658](https://stackoverflow.com/questions/25139265/opening-app-on-mac-os-x-10-8-5-results-in-error-10658)

**Comment by betabrain** [comment]
The jar version works flawlessly.

**Comment by oakes** [comment]
Not sure how to address this because I can't reproduce it. Your java version shouldn't matter since the app contains its own copy of the JRE. If anyone else has this issue please let me know here.

**Comment by hellonico** [comment]
On Sierra, I cannot reproduce this either. Double clicking on the .app file or trying to see if any of those links ideas can help: 
- http://apple.stackexchange.com/questions/224003/lsopenurlswithrole-failed-for-the-application-applications-darktable-app-with
- http://alvinalexander.com/scala/lsopenurlswithrole-failed-with-error-solution-mac-osx
- http://apple.stackexchange.com/questions/181026/lsopenurlswithrole-failed-with-error-10810-cant-open-install-os-x-yosemite

**Comment by betabrain** [comment]
**Inspired by the first link**, I ran `opensnoop`, but to be honest I am quite unfamiliar with dtrace especially concerning on OSX a combination Google doesn't know all about.

```
  UID  PID   COMM          FD PATH
  501  23747 Nightcode      3 /dev/dtracehelper    
  501  23747 Nightcode      3 /Applications/Nightcode.app 
  501  23747 Nightcode      3 /Applications/Nightcode.app/Contents 
  501  23747 Nightcode      3 /Applications/Nightcode.app/Contents/Info.plist 
  501  23747 Nightcode      3 /Applications/Nightcode.app/Contents/MacOS/Nightcode 
  501  23747 Nightcode     -1 /usr/share/locale/en_US.utf8/LC_COLLATE 
  501  23747 Nightcode      3 /Applications/Nightcode.app/Contents/MacOS/libpackager.dylib 
  501  23747 Nightcode     -1 /etc/.mdns_debug     
  501  23747 Nightcode      3 /etc/localtime
```
Not quite sure why there attempts to access files that don't exist. `/etc/.mdns_debug` seems especially weird. That said, lots of apps come with frameworks/libraries that try to load stuff, but fail gracefully.

I also got lots of lines like the following, but to be honest I don't know how to interpret it. Are OSX's dtrace scripts fragile?
```
dtrace: error on enabled probe ID 5 (ID 152: syscall::open:return): invalid user access in action #11 at DIF offset 24
dtrace: error on enabled probe ID 6 (ID 938: syscall::open_nocancel:return): invalid user access in action #11 at DIF offset 24
```
**The second link** mentioned missing jars and suggested looking at `/var/log/system.log`:
```
Nov 16 05:41:18 foggy Nightcode[23903]: Failed to find library:
Nov 16 05:41:18 foggy Nightcode[23903]: .:Failed to launch JVM
Nov 16 05:41:18 foggy com.apple.xpc.launchd[1] (com.apple.xpc.launchd.oneshot.0x10000017.Nightcode[23903]): Service exited with abnormal code: 1
```
I doubt it's a missing jar issue, because they are all specified in the project - I can build a functioning ueberjar without issues. Maybe it's the bundled JRE?

**About the third link:** I tried `chmod +x` and reinstalled from a freshly downloaded `.dmg` just, but no changes.

**Comment by betabrain** [comment]
Unless someone looks at this and immediately knows what's up, we better stop here. Going down this rabbit hole eats too much time. I am happy running from the jar for now.

It's so unfortunate that computers are not handling state from the ground up in a Clojure or even Erlang fashion. 😢

**Comment by oakes** [comment]
The latest release bundles a newer version of the JVM so it may be worth trying again. I'll close for now, let me know if you think this may still be a bug on my end.

**Comment by rostopira** [comment]
Same bug on El Capitan with v2.2.2

**Comment by oakes** [comment]
Gotcha, I'll re-open this then.

**Comment by allllan** [comment]
Info.plist references the plugins directory as 'plugins' -- all lower case. The actual directory is called 'PlugIns'. If you have a case sensitive file system, try creating a symlink from plugins to PlugIns. This solved the issue for me.

  <string>$APPDIR/plugins/Java</string>

**Comment by oakes** [comment]
@allllan looks like you found the issue! That file is created by the `javapackager` tool, so I suppose that is the source of the bug.

**Comment by oakes** [comment]
I released 2.4.0 with JRE 9, and it looks like the Info.plist doesn't even mention plugins, so I am guessing it now works on case-sensitive file systems.
